code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -16,517 +16,23 @@
 /obj/item/weapon/reagent_containers/food/drinks/drinkingglass/on_reagent_change()
 	cut_overlays()
 	if (reagents.reagent_list.len > 0)
```

Comment:
```
/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/on_reagent_change()
	cut_overlays()
	if(reagents.reagent_list.len)
		var/datum/reagent/R = reagents.get_master_reagent()
		name = R.drink_name
		desc = R.drink_desc
		if(R.drink_icon)
			icon_state = R.drink_icon
		else
			var/image/I = image(icon, "glassoverlay")
			I.color = mix_color_from_reagents(reagents.reagent_list)
			add_overlay(I)		
	else
		icon_state = "glass_empty"
		name = "glass"
		desc = "Your standard drinking glass."
```

Then for the reagents:

```
	var/drink_icon = null
	var/drink_name = "Glass of ..what?"
	var/drink_desc = "You can't really tell what this is."
```


This is much cleaner. Basically, it'll do the same thing, only if there is no drink icon, it'll mix the colors from the contained reagents